src/catalog/events/src/main/java/org/geoserver/cloud/config/catalog/events/GeoServerLifecycleEventPublisher.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * (c) 2024 Open Source Geospatial Foundation - all rights reserved This code is licensed under the
+ * GPL 2.0 license, available at the root application directory.
+ */
+package org.geoserver.cloud.config.catalog.events;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.geoserver.cloud.event.lifecycle.LifecycleEvent;
+import org.geoserver.cloud.event.lifecycle.ReloadEvent;
+import org.geoserver.cloud.event.lifecycle.ResetEvent;
+import org.geoserver.config.impl.GeoServerLifecycleHandler;
+
+import java.util.function.Consumer;
+
+@RequiredArgsConstructor
+@Slf4j
+class GeoServerLifecycleEventPublisher implements GeoServerLifecycleHandler {
```

Comment:
Can you add some class javadoc explaining what it does?

Reply:
Done in d0d82ee1c1d38c2d2f4f09954df7239cd60b7089